keadm/cmd/keadm/app/cmd/edge/join_others.go

```diff
@@ -181,6 +184,12 @@ func createEdgeConfigFiles(opt *common.JoinOptions) error {
 }
 
 func join(opt *common.JoinOptions, step *common.Step) error {
+	if opt.PreRun != "" {
+		step.Printf("Execute the prescript before joining the node")
+		if err := util.RunScript(opt.PreRun); err != nil {
+			return err
+		}
+	}
```

Comment:
Move to PreRunE，and add `--post-run` flag. Reset need add `--pre-rune` flag too.
```goalng
PreRunE: func(cmd *cobra.Command, args []string) error {
    ...
    if opt.PreRun != "" {
        // run script
    }
}
PostRun: func(cmd *cobra.Command, args []string) error {
    ....
    if opt.PostRun != "" {
        // run script without err handle
    }
}
```

Reply:
Thank you for your suggestion. I have modified it.